src/pyfmi/fmi.pyx

```diff
@@ -27,8710 +27,89 @@ For profiling:
-            self._derivatives_inputs_dependencies is not None):
-               return self._derivatives_states_dependencies, self._derivatives_inputs_dependencies
-
-        cdef size_t *dependencyp
-        cdef size_t *start_indexp
-        cdef char   *factor_kindp
-        cdef FMIL.fmi2_import_variable_t *variable
-        cdef FMIL.fmi2_import_variable_list_t *variable_list
-
-        derivatives = list(self.get_derivatives_list().keys())
-        states_list = list(self.get_states_list().keys())
-        inputs_list = list(self.get_input_list().keys())
-
-        states = OrderedDict()
-        states_kind = OrderedDict()
-        inputs = OrderedDict()
-        inputs_kind = OrderedDict()
-
-        if len(derivatives) != 0:
-            FMIL.fmi2_import_get_derivatives_dependencies(self._fmu, &start_indexp, &dependencyp, &factor_kindp)
-
-            if start_indexp == NULL:
-                logging.warning(
-                    'No dependency information for the derivatives was found in the model description.' \
-                    ' Assuming complete dependency.')
-                for i in range(0,len(derivatives)):
-                    states[derivatives[i]]  = states_list
-                    states_kind[derivatives[i]] = [FMI2_KIND_DEPENDENT]*len(states_list)
-
-                    inputs[derivatives[i]]  = inputs_list
-                    inputs_kind[derivatives[i]] = [FMI2_KIND_DEPENDENT]*len(inputs_list)
-            else:
-                variable_list = FMIL.fmi2_import_get_variable_list(self._fmu, 0)
-                if variable_list == NULL:
-                    raise FMUException("The returned variable list is NULL.")
-
-                for i in range(0,len(derivatives)):
-                    states[derivatives[i]]  = []
-                    states_kind[derivatives[i]]  = []
-
-                    inputs[derivatives[i]] = []
-                    inputs_kind[derivatives[i]] = []
-
-                    for j in range(0, start_indexp[i+1]-start_indexp[i]):
-                        if dependencyp[start_indexp[i]+j] != 0:
-                            variable = FMIL.fmi2_import_get_variable(variable_list, dependencyp[start_indexp[i]+j]-1)
-                            name             = decode(FMIL.fmi2_import_get_variable_name(variable))
-
-                            if name in states_list:
-                                states[derivatives[i]].append(name)
-                                states_kind[derivatives[i]].append(factor_kindp[start_indexp[i]+j])
-
-                            elif name in inputs_list:
-                                inputs[derivatives[i]].append(name)
-                                inputs_kind[derivatives[i]].append(factor_kindp[start_indexp[i]+j])
-
-                #Free the variable list
-                FMIL.fmi2_import_free_variable_list(variable_list)
-
-        #Caching
-        self._derivatives_states_dependencies = states
-        self._derivatives_states_dependencies_kind = states
```

Comment:
Essentially all the imports for backward compatibility.

Reply:
Can we put each import on a new row? For example
```
from a import (
    b,
    c,
    ...
)
```

Reply:
Fixed.